src/pstore/pstore.c

```diff
@@ -170,8 +170,8 @@ static int move_file(PStoreEntry *pe, const char *subdir) {
 }
 
 static int write_dmesg(const char *dmesg, size_t size, const char *id) {
-        _cleanup_(unlink_and_freep) char *ofd_path = NULL;
-        _cleanup_free_ char *tmp_path = NULL;
+        _cleanup_(unlink_and_freep) char *tmp_path = NULL;
+        _cleanup_free_ char *ofd_path = NULL;
```

Comment:
the `ofd_path = mfree(ofd_path)` line at the end of this function also needs updating

Reply:
(it should be changed to `tmp_path = mfree(tmp_path)` to "disarm" the unlink destructor, since at that point the file is already gone on disk

Reply:
I see. Fixed.